File: TPS/TP02/Q01-Class-Show-JAVA/Show.java
import java.util.*;

class Show {
	private String show_id;
	private String type;
	private String title;
	private String director;
	private String cast[];
	private String country;
	private Date date_added;
	private int release_year;
	private String rating;
	private String duration;
	private String listed_in[];

	public Show() {
		setId("NaN");
		setType("NaN");
		setTitle("NaN");
		setDirector("NaN");
		setCast("NaN";
		setCountry("NaN");
		setDate("NaN");
		setReleaseYear("NaN");
		setRating("NaN");
		setDuration("NaN");
		setListedIn("NaN");
	}

	public Show(String id, String type, String title, String director, String cast[], String country,
				Date date, int release_year, String rating, String duration, String listed_in[]) {
		setId(id);
		setType(type);
		setTitle(title);
		setDirector(director);
		setCast(cast);
		setCountry(country);
		setDate(date);
		setReleaseYear(release_year);
		setRating(rating);
		setDuration(duration);
		setListedIn(listed_in);
		int n = 0;
	}
	
	public void setListedIn(String listed_in[]) {

	}

	public void setDuration(String duration) {
		this.duration = duration;
	}

	public void setRating(String rating) {
		this.rating = rating;
	}

	public void setReleaseYear(int release_year) {
		this.release_year = release_year;
	}

	public void setDate(Date date) {

	}

	public void setCountry(String country) {
		this.country = country;
	}
	
	public void setCast(String cast) {
	
	}

	private void setDirector(String director) {
		this.director = director;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public void setType(String type) {
		this.type = type;
	}

	public void setId(String id) {
		this.show_id = id;
	}

	public String getShowId() {
		return this.show_id;
	}

	public String getType() {
		return this.type;
	}

	public String getTitle() {
		return this.title;
	}

	public String getDirector() {
		return this.director;
	}

	public String[] getCast() {
		return this.cast;
	}

	public String getCountry() {
		return this.country;
	}

	public Date getDate() {
		return this.date_added;
	}

	public int getReleaseYear() {
		return this.release_year;
	}

	public String getRating() {
		return this.rating;
	}

	public String getDuration() {
		return this.duration;
	}

	public String[] getListedIn() {
		return this.listed_in;
	}
}

File: TPS/TP02/Q02-Class-Show-C/show.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <stdbool.h>

typedef struct {
    char showId[];
    char type[];
    char title[];
    char director[];
    char cast[];
    char country[];
    Date dateAdded;
    int releaseYear;
    char rating[];
    char duration[];
    char listedIn[];
} Show;

void setShowId(Show *show, char *id) {
    strcpy(show->showId, id);
}

void setType(Show *show, char *type) {
    strcpy(show->type, type);
}

void setTitle(Show *show, char *title) {
    strcpy(show->title, title);
}

void setDirector(Show *show, char *director) {
    strcpy(show->director, director);
}

void setCast(Show *show, char *cast) {
    strcpy(show->cast, cast);
}

void setCountry(Show *show, char *country) {
    strcpy(show->country, country);
}

void setDateAdded(Show *show, Date *dateAdded) {
    show->dateAdded = *dateAdded;
}

void setReleaseYear(Show *show, int *releaseYear) {
    strcpy(show->releaseYear, releaseYear);
}

void setRating(Show *show, char *rating) {
    strcpy(show->rating, rating);
}

void setDuration(Show *show, char *duration) {
    strcpy(show->duration, duration);
}

void setListedIn(Show *show, char *listedIn) {
    strcpy(show->listedIn, listedIn);
}

char[] getShowId(Show *show) {
    return show->showId;
}

char[] getType(Show *show) {
    return show->type;
}

char[] getTitle(Show *show) {
    return show->title;
}

char[] getDirector(Show *show) {
    return show->director;
}

char[] getCast(Show *show) {
    return show->cast;
}

char[] getCountry(Show *show) {
    return show->country;
}

Date getDateAdded(Show *show) {
    return show->dateAdded;
}

int getReleaseYear(Show *show) {
    return show->releaseYear;
}

char[] getRating(Show *show) {
    return show->rating;
}

char[] getDuration(Show *show) {
    return show->duration;
}

char[] getListedIn(Show *show) {
    return show->listedIn;
}

void imprimir(Show* show) {
    printf("=> %s ", getShowId(show));
    printf("## %s ", getType(show));
    printf("## %s ", getTitle(show));
    printf("## %s ", getDirector(show));
    printf("## %s ", getCast(show));
    printf("## %s ", getCountry(show));
    printf("## %d/%d/%d ", show->dateAdded.day, show->dateAdded.month, show->dateAdded.year);
    printf("## %d ", getReleaseYear(show));
    printf("## %s ", getRating(show));
    printf("## %s ", getDuration(show));
    printf("## %s ", getListedIn(show));
}

